packages/vm/src/imports.rs

```diff
@@ -171,13 +171,25 @@ pub fn do_write<S: Storage, Q: Querier>(
     key_ptr: u32,
     value_ptr: u32,
 ) -> VmResult<i32> {
+    if is_storage_readonly::<S, Q>(ctx) {
+        return Err(make_generic_err(
+            "Writing to storage is prohibited in this context.",
+        ));
+    }
+
```

Comment:
Wouldn't we want to have a specific error variant for this case?

Reply:
Thanks, added in 66d9ccf0111ce0bb61f4d38700eaca3fdd09b23f